pkg/ossm/operator.go

```diff
@@ -0,0 +1,139 @@
+// limitations under the License.
+
+package ossm
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/maistra/maistra-test-tool/pkg/util"
+)
+
+func cleanupOperatorTest() {
+	util.Log.Info("Cleanup ...")
+	//delete the infra label from the worker node
+	util.Log.Info("Delete Nodeselector and Tolerations from the operator")
+	_, err := util.Shell(`oc patch subscription servicemeshoperator -n openshift-operators --type json -p='[{"op": "remove", "path": "/spec/config/nodeSelector"}]'`)
+	if err != nil {
+		util.Log.Error("Failed to remove the nodeSelector from the operator")
+	}
+	_, err = util.Shell(`oc patch subscription servicemeshoperator -n openshift-operators --type json -p='[{"op": "remove", "path": "/spec/config/tolerations"}]'`)
+	if err != nil {
+		util.Log.Error("Failed to remove the tolerations from the operator")
+	}
+	util.Log.Info("Operator patching completed")
+	response, err := util.Shell(`oc adm taint nodes -l node-role.kubernetes.io/infra node-role.kubernetes.io/infra=reserved:NoSchedule- node-role.kubernetes.io/infra=reserved:NoExecute-`)
+	if err != nil {
+		util.Log.Error("Failed to taint the infra node")
+	}
+	util.Log.Info(response)
+	if strings.Contains(response, "untainted") {
+		util.Log.Info("Infra node is not tainted anymore")
+	} else {
+		util.Log.Error("Fail to remove the taint from the infra node")
+	}
+
+}
+
+// TestOperator tests scenario to cover all the test cases related to the OSSM operators
+func TestOperator(t *testing.T) {
+	defer cleanupOperatorTest()
+	defer util.RecoverPanic(t)
+
+	util.Log.Info("Test cases related to OSSM Operators")
+
+	//test case regarding the https://issues.redhat.com/browse/OSSM-2342 issue to run OSSM operator on infra nodes
+	t.Run("test_ossm_operator_deploy_infra_nodes", func(t *testing.T) {
+		defer util.RecoverPanic(t)
+		util.Log.Info("Testing: Run OSSM Operator on infra nodes")
+		//Get and pick one worker node that does not have already installed the istio operator
+		workername := pickWorkerNode(t)
+		//Label the worker node as infra
+		_, err := util.Shell(`oc label node %s node-role.kubernetes.io/infra=`, workername)
+		if err != nil {
+			t.Fatalf("Failed to label the worker node as infra")
+		}
+		_, err = util.Shell(`oc label node %s node-role.kubernetes.io=infra`, workername)
+		if err != nil {
+			t.Fatalf("Failed to label the worker node as infra")
+		}
+		//verify that the worker node is labeled as infra
+		name, err := util.Shell(`oc get nodes -l node-role.kubernetes.io/infra= -o jsonpath='{.items[0].metadata.name}'`)
+		if err != nil {
+			t.Fatalf("Failed to get the infra node name")
+		}
+		if name != workername {
+			t.Fatalf("Failed to label the worker node as infra")
+		}
+		//Taint the node. The only validation to check this is the output message
+		response, err := util.Shell(`oc adm taint nodes -l node-role.kubernetes.io/infra node-role.kubernetes.io/infra=reserved:NoSchedule node-role.kubernetes.io/infra=reserved:NoExecute`)
```

Comment:
Same question above. I think this new test case is only applicable for running as a cluster-admin user e.g. kubeadmin. 
Hi @FilipB  , do you know if our jenkins user is a regular or cluster-admin user when running maistra-test-tool ? If it's a regular user, this new case may not work when running oc adm commends.

Reply:
Tested this too, I run the test from local having the oc login with the jenkins user and the oc adm taint nodes works perfectly:
`INFO[2023-03-13T11:54:10+01:00] Running command oc adm taint nodes -l node-role.kubernetes.io/infra node-role.kubernetes.io/infra=reserved:NoSchedule node-role.kubernetes.io/infra=reserved:NoExecute
INFO[2023-03-13T11:54:11+01:00] Command output:
node/fhocp413-mvrbv-worker-0-8k9hw tainted`